src/monattack.cpp

```diff
@@ -983,6 +983,19 @@ bool mattack::resurrect( monster *z )
     return true;
 }
 
+void mattack::smash_specific( monster *z, Creature *target )
+{
+    if( target == nullptr || !is_adjacent( z, target, false ) ) {
+        return;
+    }
+    if( z->has_flag( MF_RIDEABLE_MECH ) ) {
+        z->mod_mech_power( -5 );
+    }
+    tripoint targ_pos = target->pos();
+    z->set_goal( targ_pos );
```

Comment:
```suggestion
    z->set_goal( target->pos() );
```

Reply:
Has some `const` problems if I use target->pos() directly...

Reply:
Now you can do this.